src/main/java/org/wensheng/juicyraspberrypie/JuicyRaspberryPie.java

```diff
@@ -21,187 +21,183 @@
+		}
+	}
+
+	@Override
+	@SuppressWarnings("PMD.DoNotUseThreads")
+	public void onEnable() {
+		this.saveDefaultConfig();
+		final int port = this.getConfig().getInt("api_port");
+
+		//create new tcp listener thread
+		try {
+			serverThread = new ServerListenerThread(this, new InetSocketAddress(port));
+			new Thread(serverThread).start();
+			logger.info("ThreadListener Started");
+		} catch (final IOException e) {
+			logger.log(Level.WARNING, "Failed to start ThreadListener", e);
+			return;
+		}
+		getServer().getScheduler().scheduleSyncRepeatingTask(this, new TickHandler(), 1, 1);
+		final boolean startPyserver = this.getConfig().getBoolean("start_cmdsvr");
+
+		this.saveResources();
+
+		if (startPyserver) {
+			final String pyexe = getConfig().getString("pyexe", "python.exe");
+			logger.info("Starting Python command server using " + pyexe);
+			final ProcessBuilder processBuilder = new ProcessBuilder(pyexe, "cmdsvr/pycmdsvr.py");
+			try {
+				processBuilder.redirectErrorStream(true);
+				processBuilder.directory(this.getDataFolder());
+				processBuilder.start();
+			} catch (final IOException e) {
+				logger.warning("******************************************************************************");
+				logger.warning("Could not start python command server! Please check your `pyexe` in config.yml");
+				logger.warning("******************************************************************************");
+			}
+		}
+
+	}
+
+	@Override
+	public void onDisable() {
+		int port = this.getConfig().getInt("cmdsvr_port");
+		// cmdsvr_host is not used, always "localhost" for now
+		final boolean startPyserver = this.getConfig().getBoolean("start_cmdsvr");
+		if (port == 0) {
+			port = 32_123;
+		}
+
+		if (startPyserver) {
+			try (Socket socket = new Socket("localhost", port);
+				 DataOutputStream toPyServer = new DataOutputStream(socket.getOutputStream())) {
+				toPyServer.writeUTF("BYE");
+				logger.info("ask py server to shut itself down");
+			} catch (final IOException e) {
+				logger.warning("Could not send shutdown signal to python command server, please shutdown manually.");
+			}
+		}
+
+		getServer().getScheduler().cancelTasks(this);
+		for (final RemoteSession session : sessions) {
+			session.close();
+		}
+		serverThread.close();
+
+		serverThread = null;
+	}
+
+	@Override
+	@SuppressWarnings("PMD.AvoidLiteralsInIfCondition")
+	public boolean onCommand(@NotNull final CommandSender sender, @NotNull final Command command, @NotNull final String label, final String[] args) {
+		if (args.length == 0) {
+			return false;
+		}
+
+		int port = this.getConfig().getInt("cmdsvr_port");
+		if (port == 0) {
+			port = 4731;
+		}
+
+		final String cmdString;
+		try (Socket socket = new Socket("localhost", port);
+			 DataOutputStream toPyServer = new DataOutputStream(socket.getOutputStream());
+			 BufferedReader fromPyServer = new BufferedReader(new InputStreamReader(socket.getInputStream(), StandardCharsets.UTF_8))) {
```

Comment:
Looksl ike there are some spaces here in front after tha tabs

Reply:
Thats true, its becuase it is indented like the line above, because it's code that belongs into the try(...) statement

Reply:
Ah, I see. Hm, that would break if someone has a tab size that is not 4 so I would expect there to only be two tabs and the rest spaces to keep it in line with the `(` but not sure how this is done normally when formatting such a case. (Definitely extremely minor though, the rest looks pretty good so feel free to resolve that if you don't think it needs to be changed)

Reply:
I'm not a fan of such high-maintenance styles (aligning a long parameter list after the `(` following the method name is another one; Robert did that a lot). You could add a line break after the `try (` and then have alignment with just regular tabs.